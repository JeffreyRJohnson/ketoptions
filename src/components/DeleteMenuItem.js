import React from 'react';

const DeleteMenuItem = () => <div>This is my delete Menu item component</div>;

export default DeleteMenuItem;
