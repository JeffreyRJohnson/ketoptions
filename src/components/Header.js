import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1 className="header">Ketoptions</h1>
    <NavLink
      to="/"
      activeClassName="is-active"
      exact={true}
      className="header__link"
    >
      Landing Page
    </NavLink>
    <NavLink
      to="/dashboard"
      activeClassName="is-active"
      className="header__link"
    >
      Dashboard
    </NavLink>
    <NavLink to="/create" activeClassName="is-active" className="header__link">
      Create
    </NavLink>
    <NavLink to="/edit" activeClassName="is-active" className="header__link">
      Edit
    </NavLink>
    <NavLink to="/delete" activeClassName="is-active" className="header__link">
      Delete
    </NavLink>
  </header>
);

export default Header;
