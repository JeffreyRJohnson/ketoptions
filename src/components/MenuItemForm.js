import React from 'react';

export default class MenuItemForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      menu_item: props.menuItem ? props.menuItem.menu_item : '',
      restaurant_name: props.menuItem ? props.menuItem.restaurant_name : '',
      calories: props.menuItem ? props.menuItem.calories : '',
      protein: props.menuItem ? props.menuItem.protein : '',
      fat: props.menuItem ? props.menuItem.fat : '',
      carbs: props.menuItem ? props.menuItem.carbs : '',
      error: ''
    };
  }

  onRestaurantNameChange = e => {
    const restaurant_name = e.target.value;
    this.setState(() => ({ restaurant_name }));
  };

  onMenuItemChange = e => {
    const menu_item = e.target.value;
    this.setState(() => ({ menu_item }));
  };

  onCalorieChange = e => {
    const calories = e.target.value;
    this.setState(() => ({ calories }));
  };

  onProteinChange = e => {
    const protein = e.target.value;
    this.setState(() => ({ protein }));
  };

  onFatChange = e => {
    const fat = e.target.value;
    this.setState(() => ({ fat }));
  };

  onCarbChange = e => {
    const carbs = e.target.value;
    this.setState(() => ({ carbs }));
  };

  onSubmit = e => {
    e.preventDefault();

    if (
      !this.state.restaurant_name ||
      !this.state.menu_item ||
      !this.state.calories ||
      !this.state.protein ||
      !this.state.fat ||
      !this.state.carbs
    ) {
      this.setState(() => ({ error: 'You are missing a field' }));
    } else {
      this.setState(() => ({ error: '' }));
      this.props.onSubmit({
        restaurant_name: this.state.restaurant_name,
        menu_item: this.state.menu_item,
        calories: this.state.calories,
        protein: this.state.protein,
        fat: this.state.fat,
        carbs: this.state.carbs
      });
    }
  };

  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>
          <input
            type="text"
            placeholder="Restaurant name"
            autoFocus
            value={this.state.restaurant_name}
            onChange={this.onRestaurantNameChange}
          />
          <input
            type="text"
            placeholder="Menu Item name"
            value={this.state.menu_item}
            onChange={this.onMenuItemChange}
          />
          <input
            type="number"
            placeholder="Calories"
            value={this.state.calories}
            onChange={this.onCalorieChange}
          />
          <input
            type="number"
            placeholder="Protein"
            value={this.state.protein}
            onChange={this.onProteinChange}
          />
          <input
            type="number"
            placeholder="Fat"
            value={this.state.fat}
            onChange={this.onFatChange}
          />
          <input
            type="number"
            placeholder="Carbs"
            value={this.state.carbs}
            onChange={this.onCarbChange}
          />
          <button>Add Menu Item</button>
        </form>
      </div>
    );
  }
}
