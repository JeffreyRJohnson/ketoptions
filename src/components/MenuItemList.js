import React from 'react';
import { connect } from 'react-redux';
import MenuItemListItem from './MenuItemListItem';
import selectMenuItems from '../selectors/menuItems';

export const MenuItemList = props => (
  <div>
    {props.menuItems.length === 0 ? (
      <p>No menu items</p>
    ) : (
      props.menuItems.map(menuItem => {
        return <MenuItemListItem key={menuItem.id} {...menuItem} />;
      })
    )}
  </div>
);

const mapStateToProps = state => {
  return {
    menuItems: selectMenuItems(state.menuItems, state.filters)
  };
};

export default connect(mapStateToProps)(MenuItemList);
