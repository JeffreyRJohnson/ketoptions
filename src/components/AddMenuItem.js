import React from 'react';
import { connect } from 'react-redux';
import MenuItemForm from './MenuItemForm';
import { startAddMenuItem } from '../actions/menuItems';

export class AddMenuItem extends React.Component {
  onSubmit = menuItem => {
    this.props.startAddMenuItem(menuItem);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <h1>Add Menu Item</h1>
        <MenuItemForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startAddMenuItem: menuItem => dispatch(startAddMenuItem(menuItem))
});

export default connect(
  undefined,
  mapDispatchToProps
)(AddMenuItem);
