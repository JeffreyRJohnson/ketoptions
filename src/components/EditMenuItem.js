import React from 'react';
import { connect } from 'react-redux';
import MenuItemForm from './MenuItemForm';
import { editMenuItem, removeMenuItem } from '../actions/menuItems';

export class EditMenuItem extends React.Component {
  onSubmit = menuItem => {
    this.props.editMenuItem(this.props.menuItem.id, menuItem);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.removeMenuItem({ id: this.props.menuItem.id });
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <MenuItemForm menuItem={this.props.menuItem} onSubmit={this.onSubmit} />
        <button onClick={this.onRemove}>Remove</button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  menuItem: state.menuItems.find(
    menuItem => menuItem.id === props.match.params.id
  )
});

const mapDispatchToProps = dispatch => ({
  editExpense: (id, menuItem) => dispatch(editMenuItem(id, menuItem)),
  removeMenuItem: data => dispatch(removeMenuItem(data))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditMenuItem);
