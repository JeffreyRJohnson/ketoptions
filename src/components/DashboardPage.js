import React from 'react';
import MenuItemList from './MenuItemList';
import MenuItemListFilters from './MenuItemListFilters';

const DashboardPage = () => (
  <div>
    <MenuItemListFilters />
    <MenuItemList />
  </div>
);

export default DashboardPage;
