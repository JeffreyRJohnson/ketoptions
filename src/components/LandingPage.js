import React from 'react';
import MenuItemList from './MenuItemList';
import MenuItemListFilters from './MenuItemListFilters';

const LandingPage = () => (
  <div>
    <MenuItemListFilters />
    <MenuItemList />
  </div>
);

export default LandingPage;
