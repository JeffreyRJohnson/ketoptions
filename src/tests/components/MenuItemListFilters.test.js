import React from 'react';
import { shallow } from 'enzyme';
import { MenuItemListFilters } from '../../components/MenuItemListFilters';
import { filters, altFilters } from '../fixtures/filters';

let setTextFilter, wrapper;

beforeEach(() => {
  setTextFilter = jest.fn();
  wrapper = shallow(
    <MenuItemListFilters filters={filters} setTextFilter={setTextFilter} />
  );
});

test('should render MenuItemListFilters correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render MenuItemListFilters with alt data correctly', () => {
  wrapper.setProps({
    filters: altFilters
  });
  expect(wrapper).toMatchSnapshot();
});

test('should handle text change', () => {
  const value = 'Wendys';
  wrapper.find('input').simulate('change', {
    target: { value }
  });
  expect(setTextFilter).toHaveBeenLastCalledWith(value);
});
