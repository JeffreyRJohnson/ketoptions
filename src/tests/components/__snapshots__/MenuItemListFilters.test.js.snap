// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MenuItemListFilters correctly 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`should render MenuItemListFilters with alt data correctly 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text"
    value="McDonalds"
  />
</div>
`;
